src/python/WMCore/ReqMgr/Service/RequestAdditionalInfo.py

```diff
@@ -6,11 +6,72 @@
 from WMCore.ReqMgr.Utils.Validation import get_request_template_from_type
 
 import WMCore.ReqMgr.Service.RegExp as rx
-from WMCore.REST.Format import JSONFormat
+from WMCore.REST.Format import JSONFormat, PrettyJSONFormat
 
 from WMCore.Wrappers import JsonWrapper
 from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
 
+#TODO: need to get algorithm proper way
+ALGO_DICT = {"FileBased": {"files_per_job": ''},
+             "TwoFileBased": {"two_files_per_job": ''},
+             "LumiBased": {"lumis_per_job": '', 
+                          "halt_job_on_file_boundaries": True},
+             "EventAwareLumiBased": {"avg_events_per_job": '',
+                                     "max_events_per_lumi": '',
+                                     "halt_job_on_file_boundaries": True},
+             "EventBased": {"events_per_job": '',
+                            "events_per_lumi": '',
+                            "lheInputFiles": False},
+             "Harvest": {"periodic_harvest_interval": ''},
+             "ParentlessMergeBySize": {"min_merge_size": '',
```

Comment:
We shouldn't touch Merge algos. AFAIK WMCore enforces it according to the task type and job spliting for the Processing/Production task